tck/src/main/java/org/eclipse/microprofile/fault/tolerance/tck/bulkhead/BulkheadFutureTest.java

```diff
@@ -87,19 +91,18 @@ public void testBulkheadMethodAsynchFutureDoneAfterGet() {
             result = bhBeanMethodAsynchronousDefault.test(fc);
         }
         catch (InterruptedException e1) {
-            Assert.fail("Unexpected interruption", e1);
+            fail("Unexpected interruption", e1);
         }
 
-        Assert.assertFalse(result.isDone(), "Future reporting Done when not");
+        assertFalse(result.isDone(), "Future reporting Done when not");
         try {
-            String r;
-            Assert.assertTrue("GET.".equals(r = result.get()), r);
-            Assert.assertTrue("GET.GET_TO.".equals(r = result.get(1, TimeUnit.SECONDS)), r);
+            assertNotNull(result.get());
```

Comment:
Please revert this line to make sure the result is the same.

Reply:
This is the crux of this PR so if you want it reverted we can just close it.

Reply:
Could you please explain more? The future must still be completed with the same value, no?

Reply:
The asserts only make sense and hold true if delegation is used to implement the feature. Other ways of implementation will not show the behaviour expected by the asserts because the underlying solution does not use delegation to link to the asynchronously completing computation which is correct to uphold the semantics of `Future` but is not sufficient for the expectations of the test. The test makes an assumption that is beyond `Future` semantics forcing a particular solution. See also the PR description.

Reply:
So what does `Future.get()` return? `null`? Arbitrary `Object`? Anything it wants?

Reply:
Rather than changing the test to not check the result, would it make more sense to change `FutureChecker` so that it returns consistent results.

Currently it changes its result every time you call either `get()` method, which doesn't fulfil the contract of the `Future` interface.

Reply:
Whatever it returns must be consistent for `get()` and get with arguments. They cannot be different because that breaks abstraction.

Reply:
Ah, I just noticed the expected difference. That's indeed wrong. But I still think that the TCK shoudl test that the `Future` returns correct result, not just that it returns "something".